### PR TITLE
Implement assignment to selector and address in Yul

### PR DESCRIPTION
**Pull request by LucasSte**
This PR implements assignment to an external function's selector and address in Yul, using the existing infrastructure in Solang.
The logic behind this is going to be used for Borsh decoding and brought up issue #944.

### CODE REVIEW

**Review by seanyoung — CHANGES_REQUESTED** [review]
Would make sense to remove `codegen::Builtin::ExternalFunctionAddress` and `codegen::Builtin::FunctionSelector` to make this consistent? Either that or a comment that they can be removed.

**Comment by LucasSte** [comment]
> Would make sense to remove `codegen::Builtin::ExternalFunctionAddress` and `codegen::Builtin::FunctionSelector` to make this consistent? Either that or a comment that they can be removed.

Fixed! Thanks for identifying this.

**Review by seanyoung — APPROVED** [review]
Looks great, thanks!

File: src/codegen/mod.rs [hunk 1]
```diff
@@ -1197,6 +1210,36 @@ impl Expression {
             ctx,
         )
     }
+
+    fn external_function_address(&self) -> Expression {
+        if matches!(self.ty(), Type::ExternalFunction { .. }) {
```
[the remark this answers is not in the document]
Comment (LucasSte) [reply src/codegen/mod.rs:1215]: Do we need this assert in debug mode?

[the remark this answers is not in the document]
Comment (seanyoung) [reply src/codegen/mod.rs:1215]: Up to you. I'm not a fan of asserts in production code, it can really bloat the code. It does have it uses though.

**Pull request merged**